crates/bevy_ecs/src/query/state.rs

```diff
@@ -875,6 +875,63 @@ impl<D: QueryData, F: QueryFilter> QueryState<D, F> {
         }
     }
 
+    /// Return a [`PointQuery`] to get the query item for the given [`Entity`].
+    #[inline]
+    pub fn as_point_query<'w, 's>(
+        &'s mut self,
+        world: &'w mut World,
+    ) -> PointQuery<'w, 's, D::ReadOnly, F> {
+        self.update_archetypes(world);
+        let last_run = world.last_change_tick();
+        let this_run = world.read_change_tick();
+        // SAFETY: query is read only
+        unsafe {
+            self.as_readonly().as_point_query_unchecked_manual(
+                world.as_unsafe_world_cell_readonly(),
+                last_run,
+                this_run,
+            )
+        }
+    }
+
+    /// Return a [`PointQuery`] to get the query item for the given [`Entity`].
+    #[inline]
+    pub fn as_point_query_mut<'w, 's>(
+        &'s mut self,
+        world: &'w mut World,
+    ) -> PointQuery<'w, 's, D, F> {
+        self.update_archetypes(world);
+        let last_run = world.last_change_tick();
+        let this_run = world.read_change_tick();
```

Comment:
```suggestion
        let this_run = world.change_tick();
```
This should use non-atomic operations since we have mutable access to the World.